addons/mrp/tests/test_replenish.py

```diff
@@ -72,6 +72,49 @@ def test_mrp_orderpoint_leadtime(self):
         # for manufacturing delay should be taken from the bom
         self.assertEqual("4.0 days", info.wh_replenishment_option_ids.lead_time)
 
+    def test_rr_picking_type_id(self):
+        """Check manufacturing order take bom according to picking type of the rule triggered by an
+        orderpoint."""
+
+        self.product_4.route_ids = self.env.ref('mrp.route_warehouse0_manufacture')
+        self.picking_type_2 = self.warehouse_1.manu_type_id.copy({'sequence': 100})
+        self.env['mrp.bom'].create({
+            'product_id': self.product_4.id,
+            'product_tmpl_id': self.product_4.product_tmpl_id.id,
+            'product_uom_id': self.uom_unit.id,
+            'product_qty': 4.0,
+            'type': 'normal',
+            'picking_type_id': self.picking_type_2.id,
+            'bom_line_ids': [
+                (0, 0, {'product_id': self.product_2.id, 'product_qty': 2}),
+                (0, 0, {'product_id': self.product_1.id, 'product_qty': 4})
+            ]})
```

Comment:
```suggestion
            'bom_line_ids': [
                Command.create({'product_id': self.product_2.id, 'product_qty': 2}),
                Command.create({'product_id': self.product_1.id, 'product_qty': 4}),
            ]}),
```
[Commands](https://www.odoo.com/documentation/master/developer/reference/backend/orm.html#odoo.fields.Command) are a bit easier to read. :slightly_smiling_face: